description: remove pdf file permission checks
--- a/xpdf/PDFCore.cc
+++ b/xpdf/PDFCore.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 2004-2013 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 18 August 2005.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -1677,9 +1679,11 @@ GString *PDFCore::extractText(int pg, do
   int x0, y0, x1, y1, t;
   GString *s;
 
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return NULL;
   }
+#endif
   if ((page = findPage(pg))) {
     cvtUserToDev(pg, xMin, yMin, &x0, &y0);
     cvtUserToDev(pg, xMax, yMax, &x1, &y1);
--- a/xpdf/XPDFCore.cc
+++ b/xpdf/XPDFCore.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 2002-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -384,12 +386,16 @@ void XPDFCore::endSelection(int wx, int
 #ifndef NO_TEXT_SELECT
       if (selectULX != selectLRX &&
 	  selectULY != selectLRY) {
+#ifdef ENFORCE_PERMISSIONS
 	if (doc->okToCopy()) {
 	  copySelection();
 	} else {
 	  error(errNotAllowed, -1,
 		"Copying of text from this document is not allowed.");
 	}
+#else
+        copySelection();
+#endif
       }
 #endif
     }
@@ -408,9 +414,11 @@ void XPDFCore::copySelection() {
   int pg;
   double ulx, uly, lrx, lry;
 
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return;
   }
+#endif
   if (getSelection(&pg, &ulx, &uly, &lrx, &lry)) {
     //~ for multithreading: need a mutex here
     if (currentSelection) {
--- a/xpdf/XPDFViewer.cc
+++ b/xpdf/XPDFViewer.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 2002-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -3501,10 +3503,12 @@ void XPDFViewer::printPrintCbk(Widget wi
   PSOutputDev *psOut;
 
   doc = viewer->core->getDoc();
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToPrint()) {
     error(errNotAllowed, -1, "Printing this document is not allowed.");
     return;
   }
+#endif
 
   viewer->core->setBusyCursor(gTrue);
 
