description: check exit codes of system calls 
author: Michael Gilbert <michael.s.gilbert@gmail.com>
Index: xpdf/xpdf/XPDFCore.cc
===================================================================
--- xpdf.orig/xpdf/XPDFCore.cc	2012-02-24 20:21:34.000000000 -0500
+++ xpdf/xpdf/XPDFCore.cc	2012-02-24 20:21:34.000000000 -0500
@@ -480,6 +480,7 @@
   Object movieAnnot, obj1, obj2, obj3;
   GString *msg;
   int i;
+  int errcode;
 
   switch (kind = action->getKind()) {
 
@@ -569,12 +570,18 @@
       if (globalParams->getLaunchCommand()) {
 	fileName->insert(0, ' ');
 	fileName->insert(0, globalParams->getLaunchCommand());
-	system(fileName->getCString());
+	errcode = system(fileName->getCString());
+        if (errcode != 0) {
+          error( -1 , "non-zero error code returned by system call");
+        }
       } else {
 	msg = new GString("About to execute the command:\n");
 	msg->append(fileName);
 	if (doQuestionDialog("Launching external application", msg)) {
-	  system(fileName->getCString());
+	  errcode = system(fileName->getCString());
+          if (errcode != 0) {
+            error( -1 , "non-zero error code returned by system call");
+          }
 	}
 	delete msg;
       }
@@ -684,6 +691,7 @@
 void XPDFCore::runCommand(GString *cmdFmt, GString *arg) {
   GString *cmd;
   char *s;
+  int errcode;
 
   if ((s = strstr(cmdFmt->getCString(), "%s"))) {
     cmd = mungeURL(arg);
@@ -700,7 +708,10 @@
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error( -1 , "non-zero error code returned by system call");
+  }
   delete cmd;
 }
 
Index: xpdf/xpdf/XPDFViewer.cc
===================================================================
--- xpdf.orig/xpdf/XPDFViewer.cc	2012-02-24 20:21:34.000000000 -0500
+++ xpdf/xpdf/XPDFViewer.cc	2012-02-24 20:22:49.000000000 -0500
@@ -1126,6 +1126,7 @@
   char *p;
   char c0, c1;
   int i;
+  int errcode;
 
   cmd = new GString();
   fmt = args[0];
@@ -1222,7 +1223,10 @@
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error( -1, "non-zero error code return by system call");
+  }
   delete cmd;
 }
 
