description: fix printing issue
--- xpdf.orig/xpdf/XPDFViewer.cc	2013-05-31 20:05:23.626308457 +0000
+++ xpdf/xpdf/XPDFViewer.cc	2013-05-31 20:06:19.678306409 +0000
@@ -3713,7 +3713,8 @@
   {
     psOut = new PSOutputDev(psFileName->getCString(), doc, doc->getXRef(),
           doc->getCatalog(), NULL, firstPage, lastPage,
-          psModePS);
+          psModePS, globalParams->getPSPaperWidth(),
+          globalParams->getPSPaperHeight(), globalParams->getPSDuplex());
     if (psOut->isOk()) {
       doc->displayPages(psOut, firstPage, lastPage, 72, 72,
             0, gTrue, globalParams->getPSCrop(), gFalse);
@@ -3758,7 +3759,9 @@
       for (i=beginPage;; i+=step)
       {
         psOut = new PSOutputDev(psFileName->getCString(), doc, doc->getXRef(),
-              doc->getCatalog(), NULL, i, i, psModePS);
+              doc->getCatalog(), NULL, i, i, psModePS,
+	      globalParams->getPSPaperWidth(),
+              globalParams->getPSPaperHeight(), globalParams->getPSDuplex());
         if (psOut->isOk()) {
           doc->displayPages(psOut, i, i, 72, 72,
                 0, gTrue, globalParams->getPSCrop(), gFalse);
