Description: zoom in/out by keypad +/- too
Author: Kevin Ryde <user42_kevin@yahoo.com.au>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=757053

--- a/xpdf/XPDFViewer.cc
+++ b/xpdf/XPDFViewer.cc
@@ -603,6 +603,10 @@
     keyCode = xpdfKeyCodeDown;
   } else if (key >= XK_F1 && key <= XK_F35) {
     keyCode = xpdfKeyCodeF1 + (key - XK_F1);
+  } else if (key == XK_KP_Add) {
+    keyCode = '+';
+  } else if (key == XK_KP_Subtract) {
+    keyCode = '-';
   } else {
     return;
   }
