description: fix some invalid type conversions
author: Michael Gilbert <michael.s.gilbert@gmail.com> 
Index: xpdf/xpdf/GlobalParams.cc
===================================================================
--- xpdf.orig/xpdf/GlobalParams.cc	2012-02-24 19:42:47.000000000 -0500
+++ xpdf/xpdf/GlobalParams.cc	2012-02-24 19:42:50.000000000 -0500
@@ -86,9 +86,9 @@
 //------------------------------------------------------------------------
 
 static struct {
-  const char *name;
-  const char *t1FileName;
-  const char *ttFileName;
+  char *name;
+  char *t1FileName;
+  char *ttFileName;
 } displayFontTab[] = {
   {"Courier",               "n022003l.pfb", "cour.ttf"},
   {"Courier-Bold",          "n022004l.pfb", "courbd.ttf"},
@@ -2010,7 +2010,7 @@
   return s;
 }
 
-Unicode GlobalParams::mapNameToUnicode(const char *charName) {
+Unicode GlobalParams::mapNameToUnicode(char *charName) {
   // no need to lock - nameToUnicode is constant
   return nameToUnicode->lookup(charName);
 }
Index: xpdf/xpdf/GlobalParams.h
===================================================================
--- xpdf.orig/xpdf/GlobalParams.h	2012-02-24 19:42:47.000000000 -0500
+++ xpdf/xpdf/GlobalParams.h	2012-02-24 19:42:50.000000000 -0500
@@ -191,7 +191,7 @@
   CharCode getMacRomanCharCode(char *charName);
 
   GString *getBaseDir();
-  Unicode mapNameToUnicode(const char *charName);
+  Unicode mapNameToUnicode(char *charName);
   UnicodeMap *getResidentUnicodeMap(GString *encodingName);
   FILE *getUnicodeMapFile(GString *encodingName);
   FILE *findCMapFile(GString *collection, GString *cMapName);
Index: xpdf/xpdf/XPDFViewer.cc
===================================================================
--- xpdf.orig/xpdf/XPDFViewer.cc	2012-02-24 19:42:39.000000000 -0500
+++ xpdf/xpdf/XPDFViewer.cc	2012-02-24 19:42:50.000000000 -0500
@@ -136,7 +136,7 @@
 //------------------------------------------------------------------------
 
 struct ZoomMenuInfo {
-  const char *label;
+  char *label;
   double zoom;
 };
 
