description: fix continuous mode crashiness
author: Michael Gilbert <michael.s.gilbert@gmail.com>
bug-debian: http://bugs.debian.org/521381
Index: xpdf/xpdf/XPDFCore.cc
===================================================================
--- xpdf.orig/xpdf/XPDFCore.cc	2011-08-28 20:13:13.000000000 -0400
+++ xpdf/xpdf/XPDFCore.cc	2011-08-28 20:13:51.000000000 -0400
@@ -1358,7 +1358,7 @@
   }
 
   // draw the document
-  if (tile) {
+  if (tile && tile->image) {
     XPutImage(display, drawAreaWin, drawAreaGC, tile->image,
 	      xSrc, ySrc, xDest, yDest, width, height);
 
