description: fix continuous mode crashiness
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/521381

--- a/xpdf/XPDFCore.cc
+++ b/xpdf/XPDFCore.cc
@@ -1366,7 +1366,7 @@ void XPDFCore::redrawRect(PDFCoreTile *t
   }
 
   // draw the document
-  if (tile) {
+  if (tile && tile->image) {
     XPutImage(display, drawAreaWin, drawAreaGC, tile->image,
 	      xSrc, ySrc, xDest, yDest, width, height);
 
