description: check exit codes of system calls 
author: Michael Gilbert <michael.s.gilbert@gmail.com>

--- a/xpdf/XPDFCore.cc	2014-01-20 03:45:51.192064160 +0000
+++ b/xpdf/XPDFCore.cc	2014-01-20 03:45:51.192064160 +0000
@@ -479,6 +479,7 @@
   Object movieAnnot, obj1, obj2;
   GString *msg;
   int i;
+  int errcode;
 
   switch (kind = action->getKind()) {
 
@@ -568,12 +569,18 @@
       if (globalParams->getLaunchCommand()) {
 	fileName->insert(0, ' ');
 	fileName->insert(0, globalParams->getLaunchCommand());
-	system(fileName->getCString());
+	errcode = system(fileName->getCString());
+        if (errcode != 0) {
+          error(errInternal, -1 , "non-zero error code returned by system call");
+        }
       } else {
 	msg = new GString("About to execute the command:\n");
 	msg->append(fileName);
 	if (doQuestionDialog("Launching external application", msg)) {
-	  system(fileName->getCString());
+	  errcode = system(fileName->getCString());
+          if (errcode != 0) {
+            error(errInternal, -1 , "non-zero error code returned by system call");
+          }
 	}
 	delete msg;
       }
@@ -681,6 +688,7 @@
 void XPDFCore::runCommand(GString *cmdFmt, GString *arg) {
   GString *cmd;
   char *s;
+  int errcode;
 
   if ((s = strstr(cmdFmt->getCString(), "%s"))) {
     cmd = mungeURL(arg);
@@ -697,7 +705,10 @@
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error(errInternal, -1 , "non-zero error code returned by system call");
+  }
   delete cmd;
 }
 
--- a/xpdf/XPDFViewer.cc	2014-01-20 03:45:51.192064160 +0000
+++ b/xpdf/XPDFViewer.cc	2014-01-20 03:45:51.192064160 +0000
@@ -1126,6 +1126,7 @@
   char *p;
   char c0, c1;
   int i;
+  int errcode;
 
   cmd = new GString();
   fmt = args[0];
@@ -1222,7 +1223,10 @@
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error(errInternal, -1, "non-zero error code return by system call");
+  }
   delete cmd;
 }
 
