description: check exit codes of system calls 
author: Michael Gilbert <michael.s.gilbert@gmail.com>

--- a/xpdf/XPDFCore.cc
+++ b/xpdf/XPDFCore.cc
@@ -479,6 +479,7 @@ void XPDFCore::doAction(LinkAction *acti
   Object movieAnnot, obj1, obj2;
   GString *msg;
   int i;
+  int errcode;
 
   switch (kind = action->getKind()) {
 
@@ -568,12 +569,18 @@ void XPDFCore::doAction(LinkAction *acti
       if (globalParams->getLaunchCommand()) {
 	fileName->insert(0, ' ');
 	fileName->insert(0, globalParams->getLaunchCommand());
-	system(fileName->getCString());
+	errcode = system(fileName->getCString());
+        if (errcode != 0) {
+          error(errInternal, -1 , "non-zero error code returned by system call");
+        }
       } else {
 	msg = new GString("About to execute the command:\n");
 	msg->append(fileName);
 	if (doQuestionDialog("Launching external application", msg)) {
-	  system(fileName->getCString());
+	  errcode = system(fileName->getCString());
+          if (errcode != 0) {
+            error(errInternal, -1 , "non-zero error code returned by system call");
+          }
 	}
 	delete msg;
       }
@@ -689,6 +696,7 @@ void XPDFCore::doAction(LinkAction *acti
 void XPDFCore::runCommand(GString *cmdFmt, GString *arg) {
   GString *cmd;
   char *s;
+  int errcode;
 
   if ((s = strstr(cmdFmt->getCString(), "%s"))) {
     cmd = mungeURL(arg);
@@ -705,7 +713,10 @@ void XPDFCore::runCommand(GString *cmdFm
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error(errInternal, -1 , "non-zero error code returned by system call");
+  }
   delete cmd;
 }
 
--- a/xpdf/XPDFViewer.cc
+++ b/xpdf/XPDFViewer.cc
@@ -1134,6 +1134,7 @@ void XPDFViewer::cmdRun(GString *args[],
   char *p;
   char c0, c1;
   int i;
+  int errcode;
 
   cmd = new GString();
   fmt = args[0];
@@ -1230,7 +1231,10 @@ void XPDFViewer::cmdRun(GString *args[],
 #else
   cmd->append(" &");
 #endif
-  system(cmd->getCString());
+  errcode = system(cmd->getCString());
+  if (errcode != 0) {
+      error(errInternal, -1, "non-zero error code return by system call");
+  }
   delete cmd;
 }
 
