description: make filespec call compatible with poppler
origin: gentoo's xpdf-poppler 0.12 patches (http://bugs.gentoo.org/289803)

--- a/xpdf/XPDFCore.cc
+++ b/xpdf/XPDFCore.cc
@@ -24,6 +24,7 @@
 #include "GlobalParams.h"
 #include "PDFDoc.h"
 #include "Link.h"
+#include "FileSpec.h"
 #include "ErrorCodes.h"
 #include "GfxState.h"
 #include "CoreOutputDev.h"
@@ -476,7 +477,7 @@ void XPDFCore::doAction(LinkAction *acti
   GString *fileName, *fileName2;
   GString *cmd;
   GString *actionName;
-  Object movieAnnot, obj1, obj2;
+  Object movieAnnot, obj1, obj2, obj3;
   GString *msg;
   int i;
   int errcode;
@@ -656,7 +657,9 @@ void XPDFCore::doAction(LinkAction *acti
     if (movieAnnot.isDict()) {
       if (movieAnnot.dictLookup("Movie", &obj1)->isDict()) {
 	if (obj1.dictLookup("F", &obj2)) {
-	  if ((fileName = LinkAction::getFileSpecName(&obj2))) {
+	  if (getFileSpecNameForPlatform(&obj2, &obj3)) {
+            fileName = obj3.getString()->copy();
+            obj3.free();
 	    if (!isAbsolutePath(fileName->getCString()) &&
 		doc->getFileName()) {
 	      fileName2 = appendToPath(
