Description: Supply correct arguments to PSOutputDev's constructor
Author: Iain Lane <iain.lane@canonical.com>
Forwarded: no

--- a/xpdf/XPDFViewer.cc
+++ b/xpdf/XPDFViewer.cc
@@ -3721,8 +3721,14 @@ void XPDFViewer::printPrintCbk(Widget wi
   // Normal print mode
   if (printAll && !printBack)
   {
+    std::vector<int> pages;
+    pages.reserve((lastPage - firstPage) + 1);
+
+    for (int i = firstPage; i <= lastPage; ++i)
+        pages.push_back(i);
+
     psOut = new PSOutputDev(psFileName->getCString(), doc,
-          NULL, firstPage, lastPage,
+          NULL, pages,
           psModePS, xpdfParams->getPSPaperWidth(),
           xpdfParams->getPSPaperHeight(), xpdfParams->getPSDuplex());
     if (psOut->isOk()) {
@@ -3736,6 +3742,8 @@ void XPDFViewer::printPrintCbk(Widget wi
   {
     int step=1, i;
     int beginPage, endPage;
+    std::vector<int> page;
+    page.reserve(1);
 
     if (!printAll)
     {
@@ -3766,10 +3774,10 @@ void XPDFViewer::printPrintCbk(Widget wi
 
     if (firstPage<=lastPage)
     {
-      for (i=beginPage;; i+=step)
+      for (i=beginPage; page[0] = i; i+=step)
       {
         psOut = new PSOutputDev(psFileName->getCString(), doc,
-              NULL, i, i, psModePS,
+              NULL, page, psModePS,
               xpdfParams->getPSPaperWidth(),
               xpdfParams->getPSPaperHeight(), xpdfParams->getPSDuplex());
         if (psOut->isOk()) {
