description: use poppler's findtext api
author: Michael Gilbert <mgilbert@debian.org>

--- a/xpdf/PDFCore.cc
+++ b/xpdf/PDFCore.cc
@@ -1800,7 +1800,7 @@ GBool PDFCore::findU(Unicode *u, int len
     page = findPage(pg);
   }
   if (page->text->findText(u, len, startAtTop, gTrue, startAtLast, gFalse,
-			   caseSensitive, backward, wholeWord,
+			   caseSensitive, backward, gFalse,
 			   &xMin, &yMin, &xMax, &yMax)) {
     goto found;
   }
@@ -1819,7 +1819,7 @@ GBool PDFCore::findU(Unicode *u, int len
 	 pg += backward ? -1 : 1) {
       doc->displayPage(textOut, pg, 72, 72, 0, gFalse, gTrue, gFalse);
       if (textOut->findText(u, len, gTrue, gTrue, gFalse, gFalse,
-			    caseSensitive, backward, wholeWord,
+			    caseSensitive, backward, gFalse,
 			    &xMin, &yMin, &xMax, &yMax)) {
 	delete textOut;
 	goto foundPage;
@@ -1832,7 +1832,7 @@ GBool PDFCore::findU(Unicode *u, int len
 	 pg += backward ? -1 : 1) {
       doc->displayPage(textOut, pg, 72, 72, 0, gFalse, gTrue, gFalse);
       if (textOut->findText(u, len, gTrue, gTrue, gFalse, gFalse,
-			    caseSensitive, backward, wholeWord,
+			    caseSensitive, backward, gFalse,
 			    &xMin, &yMin, &xMax, &yMax)) {
 	delete textOut;
 	goto foundPage;
@@ -1854,7 +1854,7 @@ GBool PDFCore::findU(Unicode *u, int len
       yMax = selectLRY;
     }
     if (page->text->findText(u, len, gTrue, gFalse, gFalse, stopAtLast,
-			     caseSensitive, backward, wholeWord,
+			     caseSensitive, backward, gFalse,
 			     &xMin, &yMin, &xMax, &yMax)) {
       goto found;
     }
@@ -1871,7 +1871,7 @@ GBool PDFCore::findU(Unicode *u, int len
 	 gTrue);
   page = findPage(pg);
   if (!page->text->findText(u, len, gTrue, gTrue, gFalse, gFalse,
-			    caseSensitive, backward, wholeWord,
+			    caseSensitive, backward, gFalse,
 			    &xMin, &yMin, &xMax, &yMax)) {
     // this can happen if coalescing is bad
     goto notFound;
