description: provide an fopen wrapper for xpdf >= 3.03
author: Michael Gilbert <michael.s.gilbert@gmail.com>
--- xpdf-3.03.orig/xpdf/GlobalParams.cc
+++ xpdf-3.03/xpdf/GlobalParams.cc
@@ -73,6 +73,11 @@ extern XpdfPluginVecTable xpdfPluginVecT
 #  endif
 #endif
 
+// implement xpdf 3.03 fopen wrapper
+extern FILE *openFile(const char *path, const char *mode) {
+    return fopen(path, mode);
+}
+
 //------------------------------------------------------------------------
 
 #define cidToUnicodeCacheSize     4
--- xpdf-3.03.orig/xpdf/GlobalParams.h
+++ xpdf-3.03/xpdf/GlobalParams.h
@@ -39,6 +39,8 @@ class GlobalParams;
 class SysFontList;
 class Stream;
 
+extern FILE *openFile(const char *path, const char *mode);
+
 //------------------------------------------------------------------------
 
 // The global parameters object.
