description: fixes for poppler's modified psoutputdev
origin: gentoo xpdf-poppler patches 
Index: xpdf-3.02/xpdf/XPDFViewer.cc
===================================================================
--- xpdf-3.02.orig/xpdf/XPDFViewer.cc	2010-06-14 20:15:56.000000000 -0400
+++ xpdf-3.02/xpdf/XPDFViewer.cc	2010-06-14 20:28:42.000000000 -0400
@@ -3606,7 +3606,7 @@
   if (printAll && !printBack)
   {
     psOut = new PSOutputDev(psFileName->getCString(), doc->getXRef(),
-          doc->getCatalog(), firstPage, lastPage,
+          doc->getCatalog(), NULL, firstPage, lastPage,
           psModePS);
     if (psOut->isOk()) {
       doc->displayPages(psOut, firstPage, lastPage, 72, 72,
@@ -3652,7 +3652,7 @@
       for (i=beginPage;; i+=step)
       {
         psOut = new PSOutputDev(psFileName->getCString(), doc->getXRef(),
-              doc->getCatalog(), i, i, psModePS);
+              doc->getCatalog(), NULL, i, i, psModePS);
         if (psOut->isOk()) {
           doc->displayPages(psOut, i, i, 72, 72,
                 0, gTrue, globalParams->getPSCrop(), gFalse);
