description: use poppler's textoutputdev api 
author: Michael Gilbert <michael.s.gilbert@gmail.com>
--- xpdf-3.03.orig/xpdf/PDFCore.cc
+++ xpdf-3.03/xpdf/PDFCore.cc
@@ -901,7 +901,7 @@ void PDFCore::needTile(PDFCorePage *page
     page->links = doc->getLinks(page->page);
   }
   if (!page->text) {
-    if ((textOut = new TextOutputDev(NULL, gTrue, 0, gFalse, gFalse))) {
+    if ((textOut = new TextOutputDev(NULL, gTrue, gFalse, gFalse))) {
       doc->displayPage(textOut, page->page, dpi, dpi, rotate,
 		       gFalse, gTrue, gFalse);
       page->text = textOut->takeText();
@@ -1661,7 +1661,7 @@ GString *PDFCore::extractText(int pg, do
     }
     s = page->text->getText(x0, y0, x1, y1);
   } else {
-    textOut = new TextOutputDev(NULL, gTrue, 0, gFalse, gFalse);
+    textOut = new TextOutputDev(NULL, gTrue, gFalse, gFalse);
     if (textOut->isOk()) {
       doc->displayPage(textOut, pg, dpi, dpi, rotate, gFalse, gTrue, gFalse);
       textOut->cvtUserToDev(xMin, yMin, &x0, &y0);
@@ -1749,7 +1749,7 @@ GBool PDFCore::findU(Unicode *u, int len
   if (!onePageOnly) {
 
     // search following/previous pages
-    textOut = new TextOutputDev(NULL, gTrue, 0, gFalse, gFalse);
+    textOut = new TextOutputDev(NULL, gTrue, gFalse, gFalse);
     if (!textOut->isOk()) {
       delete textOut;
       goto notFound;
