description: fixes for poppler's modified psoutputdev
origin: gentoo's xpdf-poppler 0.12 patches (http://bugs.gentoo.org/289803)
Index: xpdf/xpdf/XPDFViewer.cc
===================================================================
--- xpdf.orig/xpdf/XPDFViewer.cc	2012-02-24 19:43:04.000000000 -0500
+++ xpdf/xpdf/XPDFViewer.cc	2012-02-24 19:43:09.000000000 -0500
@@ -3707,8 +3707,8 @@
   // Normal print mode
   if (printAll && !printBack)
   {
-    psOut = new PSOutputDev(psFileName->getCString(), doc->getXRef(),
-          doc->getCatalog(), firstPage, lastPage,
+    psOut = new PSOutputDev(psFileName->getCString(), doc, doc->getXRef(),
+          doc->getCatalog(), NULL, firstPage, lastPage,
           psModePS);
     if (psOut->isOk()) {
       doc->displayPages(psOut, firstPage, lastPage, 72, 72,
@@ -3753,8 +3753,8 @@
     {
       for (i=beginPage;; i+=step)
       {
-        psOut = new PSOutputDev(psFileName->getCString(), doc->getXRef(),
-              doc->getCatalog(), i, i, psModePS);
+        psOut = new PSOutputDev(psFileName->getCString(), doc, doc->getXRef(),
+              doc->getCatalog(), NULL, i, i, psModePS);
         if (psOut->isOk()) {
           doc->displayPages(psOut, i, i, 72, 72,
                 0, gTrue, globalParams->getPSCrop(), gFalse);
