description: make filespec call compatible with poppler
origin: gentoo's xpdf-poppler 0.12 patches (http://bugs.gentoo.org/289803)
Index: xpdf/xpdf/config.h
===================================================================
--- xpdf.orig/xpdf/config.h	2012-02-24 19:41:52.000000000 -0500
+++ xpdf/xpdf/config.h	2012-02-24 19:42:37.000000000 -0500
@@ -28,6 +28,7 @@
 #define supportedPDFVersionNum 1.7
 
 // copyright notice
+#undef xpdfCopyright
 #define xpdfCopyright "Copyright 1996-2011 Glyph & Cog, LLC"
 
 // Windows resource file stuff
Index: xpdf/xpdf/XPDFCore.cc
===================================================================
--- xpdf.orig/xpdf/XPDFCore.cc	2012-02-24 19:42:28.000000000 -0500
+++ xpdf/xpdf/XPDFCore.cc	2012-02-24 19:42:37.000000000 -0500
@@ -24,6 +24,7 @@
 #include "GlobalParams.h"
 #include "PDFDoc.h"
 #include "Link.h"
+#include "FileSpec.h"
 #include "ErrorCodes.h"
 #include "GfxState.h"
 #include "CoreOutputDev.h"
@@ -476,7 +477,7 @@
   GString *fileName, *fileName2;
   GString *cmd;
   GString *actionName;
-  Object movieAnnot, obj1, obj2;
+  Object movieAnnot, obj1, obj2, obj3;
   GString *msg;
   int i;
 
@@ -649,7 +650,9 @@
     if (movieAnnot.isDict()) {
       if (movieAnnot.dictLookup("Movie", &obj1)->isDict()) {
 	if (obj1.dictLookup("F", &obj2)) {
-	  if ((fileName = LinkAction::getFileSpecName(&obj2))) {
+	  if (getFileSpecNameForPlatform(&obj2, &obj3)) {
+	    fileName = obj3.getString()->copy();
+	    obj3.free();
 	    if (!isAbsolutePath(fileName->getCString())) {
 	      fileName2 = appendToPath(
 			      grabPath(doc->getFileName()->getCString()),
