description: use poppler's findfontfile api
author: Michael Gilbert <michael.s.gilbert@gmail.com>
Index: xpdf/xpdf/GlobalParams.cc
===================================================================
--- xpdf.orig/xpdf/GlobalParams.cc	2012-02-24 17:22:40.000000000 -0500
+++ xpdf/xpdf/GlobalParams.cc	2012-02-24 17:24:06.000000000 -0500
@@ -2085,7 +2085,7 @@
   return NULL;
 }
 
-GString *GlobalParams::findFontFile(GString *fontName) {
+GString *GlobalParams::findFontFile(GString *fontName, char** throwaway) {
   static const char *exts[] = { ".pfa", ".pfb", ".ttf", ".ttc" };
   GString *path, *dir;
 #ifdef WIN32
@@ -2095,6 +2095,7 @@
   FILE *f;
   int i, j;
 
+  return NULL;
   lockGlobalParams;
   if ((path = (GString *)fontFiles->lookup(fontName))) {
     path = path->copy();
Index: xpdf/xpdf/GlobalParams.h
===================================================================
--- xpdf.orig/xpdf/GlobalParams.h	2012-02-24 17:22:40.000000000 -0500
+++ xpdf/xpdf/GlobalParams.h	2012-02-24 17:22:44.000000000 -0500
@@ -196,7 +196,7 @@
   FILE *getUnicodeMapFile(GString *encodingName);
   FILE *findCMapFile(GString *collection, GString *cMapName);
   FILE *findToUnicodeFile(GString *name);
-  GString *findFontFile(GString *fontName);
+  GString *findFontFile(GString *fontName, char **throwaway);
   GString *findSystemFontFile(GString *fontName, SysFontType *type,
 			      int *fontNum);
   GString *findCCFontFile(GString *collection);
