description: use system xpdfrc file 
author: Michael Gilbert <michael.s.gilbert@gmail.com>
Index: xpdf-3.02/xpdf/config.h
===================================================================
--- xpdf-3.02.orig/xpdf/config.h	2010-08-07 00:30:29.000000000 -0400
+++ xpdf-3.02/xpdf/config.h	2010-08-07 00:33:05.000000000 -0400
@@ -62,9 +62,9 @@
 #ifdef SYSTEM_XPDFRC
 #define xpdfSysConfigFile SYSTEM_XPDFRC
 #else
-// under Windows, we get the directory with the executable and then
-// append this file name
-#define xpdfSysConfigFile "xpdfrc"
+// hardcode Debian's xpdfrc path since upstream's configure
+// is no longer used; hence this is not automatically populated
+#define xpdfSysConfigFile "/etc/xpdf/xpdfrc"
 #endif
 
 //------------------------------------------------------------------------
